docs/decisions/backend/0009-advanced-installer-design.md

```diff
@@ -0,0 +1,246 @@
+
+Associated UI Design:
+- https://www.figma.com/proto/Ntr9PttR5VrqpTRwwZHzLu/%F0%9F%93%B1-Mod-list?page-id=1826%3A5125&type=design&node-id=1826-5126&viewport=579%2C196%2C0.23&t=A6zTbYWd3oKZP1wF-1&scaling=min-zoom&starting-point-node-id=1826%3A5161
+
+## Context and Problem Statement
+
+There may be some situations where automated installation of a given mod might fail,
+or a user may want to install a mod manually.
+
+This can occur due to a variety of reasons; such as:
+- An oversight/bug with existing installer implementations.
+- An error on mod creator's part.
+- Unrecognized/non-standard packaging path.
+
+In the presence of these scenarios, the mod manager must still have a way to install any given mod.
+
+## Decision Drivers & Requirements
+
+- Certain mods may not come with a recognizable file structure.
+  - The Nexus Mods App might not know how to install them automatically.
+
+- Interoperability: The Nexus App must be able to somehow deploy ANY mod, regardless of its packaging format.
+  - Otherwise, if users end up having to deploy files manually, they may be discouraged from using the mod manager.
+
+- The Manual Installer must work with minimal game implementation code.
+  - It is considered the last line of defense; last resort.
+  - In other words, must not refuse to work in absence of support code for 'suggestions' etc.
+
+- UX: Must be Easy to Use
+
+  As part of our slogan named 'Make Modding Easy', it's important that the finished product is easy to use.
+
+  Namely, we have to consider that non-technical users may be using the app, and therefore we must make sure
+  to provide appropriate hints (suggestions) for them to deploy an unknown mod; even in presence of install
+  instructions from mod authors.
+
+- Must support arbitrary path mapping.
+
+  This system must support mapping any path from the source archive into any location in the target game directorie(s).
+
+  This includes mapping multiple files in the same folder to different locations, in case the modder provides
+  multiple files side by side when you are only supposed to pick 1.
+
+- Must recognise conflicts.
+
+  The end user using the advanced installer, might end up trying to map files in a way that could cause a conflict.
+
+  For example, the user might map a file to the output directory individually, then map an entire folder; and a file
+  within the folder conflicts with an already mapped file.
+
+## An Example of Manual Installation
+
+Some mods may have multiple variants, but don't use a declarative language like the FOMOD XML
+format for specifying choices. This tends to be more common in games with more niche communities
+and early mods for games.
+
+Imagine the following file structure (imaginary game):
+
+```text
+Red Hair
+├─ Data
+│  ├─ model.dae
+│  └─ hair.dds
+Green Hair
+├─ Data
+│  ├─ model.dae
+│  └─ hair.dds
+Blue Hair
+└─ Data
+   ├─ model.dae
+   └─ hair.dds
```

Comment:
Mo2 actually has an installer for mods packaged like this, I think it's the BAIN installer. 
It recognizes that multiple folders on the same level contain "valid" files or folder structures, then shows a simple installer ui with a checkbox for each option with a view on the side that shows any readme.txt contents inside each "option" folder.

We could add something like that in the future.

The main purpose of the manual installer is to handle any case where the archive is not recognized, so as you mentioned above, new mod type, wrongly packaged stuff, niece game, new formats, one of a kind mods etc.